### PR TITLE
Add feature for PomModuleDescriptorParser disabled handle source and javadoc artifact for speed up  …

### PR DESCRIPTION
**Pull request by qxo**
PomModuleDescriptorParser disabled handle source and javadoc artifact  by set disabledForSourcesAndJavadoc=true,

by ivysettings.xml  like:

``` xml
<ivysettings>
    <property name="disabledForSourcesAndJavadoc" value="true"/>
    ....

by System property,like -DdisabledForSourcesAndJavadoc=true
```

### CODE REVIEW

File: src/java/org/apache/ivy/plugins/parser/m2/PomModuleDescriptorParser.java [hunk 1]
```diff
@@ -306,6 +306,15 @@ private void addSourcesAndJavadocArtifactsIfPresent(PomModuleDescriptorBuilder m
             // no main artifact in pom, we don't need to search for meta artifacts
             return;
         }
+        
+        final boolean disabledForSourcesAndJavadoc = "true".equals(ivySettings.substitute("${disabledForSourcesAndJavadoc}"));
```
Comment (jsuereth) [inline src/java/org/apache/ivy/plugins/parser/m2/PomModuleDescriptorParser.java]: This looks great.  We should be able to set this in ivysettings.

Comment (jsuereth) [inline src/java/org/apache/ivy/plugins/parser/m2/PomModuleDescriptorParser.java]: In pedantic mode, can we call this "disableMavenAutoSourcesAndJavadoc"

**Comment by jsuereth** [comment]
One naming comment, then LGTM

**Comment by nlalevee** [comment]
I have implemented it with two properties, ivy.maven.sources.lookup and ivy.maven.javadoc.lookup, which defaults to true
Thanks for the PR

**Pull request closed without merging**